frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -294,9 +264,10 @@ void Visitor::checkNoDuplicateNamedArguments(const FnCall* node) {
       if (!actualNames.insert(name).second) {
         auto actual = node->actual(i);
         CHPL_ASSERT(actual);
-        error(actual, "the named argument '%s' is used more "
-                      "than once in the same function call.",
-                      name.c_str());
+        CHPL_POSTPARSE_ERR_SIMPLE(
```

Comment:
Can this `CHPL_POSTPARSE_ERR_SIMPLE` just be a non-macro mehtod on `builder` or `visitor`?

Reply:
Yes, it could since it has no need to generate the name of an error class like the general macro does. My main reason for keeping it as a macro is symmetry with the general macro (as well as all the other macros for error reporting in parser etc). Is there a reason it shouldn't stay as a macro?

Reply:
My personal take on this is "if we don't need a macro, we shouldn't use a macro". For existing symmetry, notice `Context`'s error method: it's just `error`, and not `CHPL_BASIC_CTX_ERROR`. My other reasons against it being a macro:

1. Readability (upper-snake-case is harder to read IMO)
2. Serves no purpose (a method works just as well)
3. Obfuscation of what's happening (it's a just a method call on `builder` (?), which you can't tell from the macro call)

Reply:
I agree the default presumption should be against using a macro. With those other reasons I think that's enough to break symmetry with `CHPL_POSTPARSE_REPORT` (especially considering we do have other non-macro error-reporting methods as you pointed out).

> it's just a method call on `builder` (?)

This is correct, except that getting the arguments for the method call includes another macro invocation (`CHPL_GET_ERROR`) to actually retrieve an instance of the desired error. This second macro invocation would be much harder to get around because it's what does the work of pasting together `Error##NAME__`, so it will probably stay.

I'll replace `CHPL_POSTPARSE_*_SIMPLE` with methods on `Builder` but plan to leave intact the use of macros beyond that unless you let me know you would also like those modified.

Reply:
Right, I wasn't saying we could change `CHPL_POSTPARSE_REPORT`, just the error reporting of "general" errors (which you've now done). I suspect the same can be done with general parse errors, too, but I can take care of that (and some other refactors I've been meaning to do) myself.